### PR TITLE
[Documentation] the good: branch

### LINKED ISSUE #1472: [Documentation] Write "The good: Branch"-section for contribution guidelines
**This is part of a larger issue - please read: https://github.com/kirbydesign/designsystem/issues/1393#issuecomment-802766408 before continuing!**

**Short description**

A "The good: Branch" section should be added to the contribution guidelines.  An outline already exists on the branch [`documentation/contribution-guidelines`](https://github.com/kirbydesign/designsystem/tree/documentation/contribution-guidelines) which should be expanded upon.

### PR DESCRIPTION
**Pull request by MadsBuchmann**
This PR closes #1472

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #1472

## What is the new behavior?
A section explaining or new branch naming policy has been included in the contribution guidelines. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
-* Feature branches
-* Bugfix branches
-* Enhancement branches
-* Housekeeping branches
-* Others i might have forgotten?
+The good branch only contain changes related to one issue and follows our branch name policy:
+
+* Issue type first followed by a `/`
+* Issue id second followed by a `-`
+* Name last
```
[the remark this answers is not in the document]
Comment (jkaltoft) [reply .github/CONTRIBUTING.md:367]: Nevermind. I think you've got it covered below 😊

**Review by jkaltoft — CHANGES_REQUESTED** [review]
Once again good stuff 👍🏻

One other thing I came to think of: Should we include something about _branching strategies_?

* Branch from `master` (or tracking issue branch if one exists)
* Are bug/hotfix branches any different?
    + Be verbose/explicit

File: .github/CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:367]: Restructured!  - added a single sentence regarding name, as I suspect most readers will not read the "why this branch naming policy?" (the markdown is not formatted yet in below example btw 👀)

```
362 
363 The good branch only contain changes related to one issue and follows our branch naming convention:
364 ``` 
365 
366 [issue-type]/[issue-id]-[name] 
367 ```
368 
369 Besides the structure, notice that the branch name:
370 * Is kebab cased (`kebab-casing-uses-hyphens-for-space`)
371 * Only uses lowercased letters
372 
373 Examples:
374 
375 ``` 
376 
377 documentation/1472-the-good-branch
378 ```
379 
380 ```
381 enhancement/1483-sticky-only-page-header
382 ```
383 
384 ```
385 housekeeping/1360-ci-cache-node-modules
386 ```
387 
388 
389 The issue type and id can be found on the [issues page](https://github.com/kirbydesign/designsystem/issues).
390 
391 ![](images/how-to-find-issue-type-and-id.png)
392 
393 The name part should give an idea of what the changes in the branch is related to, without having to look up the issue.                                                                                                                                                     
394                                                                                                                                                
395 You should now be ready to create a _good_ branch that follows our branch name policy. But if you are interested in why we have decided on this policy - read on.   
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
 
-<!-- We could also use githooks to enforce branch naming policies: https://itnext.io/using-git-hooks-to-enforce-branch-naming-policy-ffd81fa01e5e - perhaps have the warning throw a link to this section? -->
-
-* Guidelines for how to name branches 
-* Feature branches
-* Bugfix branches
-* Enhancement branches
-* Housekeeping branches
-* Others i might have forgotten?
+The good branch only contain changes related to one issue and follows our branch name policy:
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:363]: I've gone for _convention_ as I agree it sounds better than _policy_. I think strategy sounds a tad too grand, considering it is "just" how we name branches 😄

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
+
+[issue-type]/[issue-id]-[name] 
+```
+
+Example:
+
+``` 
+
+documentation/1472-the-good-branch
+```
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:383]: Done :D

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
+
+The issue type and id can be found on the [issues page](https://github.com/kirbydesign/designsystem/issues).
+
+![](images/how-to-find-issue-type-and-id.png)
+
+You should now be ready to create a _good_ branch that follows our branch name policy. But if you are interested in why we have decided on this policy - read on.
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:389]: https://github.com/kirbydesign/designsystem/pull/1490#discussion_r621084565

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
+* Issue type first followed by a `/`
+* Issue id second followed by a `-`
+* Name last
+* Is kebab cased (`kebab-casing-uses-hyphens-for-space`)
+* Lowercased letters only
+
+Template:
+
+``` 
+
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:374]: If it's 374 you're referring to, then nope 😄
Fixed with new commits!

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
+[issue-type]/[issue-id]-[name] 
+```
+
+Example:
+
+``` 
+
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:381]: Nope - should be fixed with newest commits :D

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
+documentation/1472-the-good-branch
+```
+
+The issue type and id can be found on the [issues page](https://github.com/kirbydesign/designsystem/issues).
+
+![](images/how-to-find-issue-type-and-id.png)
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:387]: Guess I better practice what I preach in the good issue 😅

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -360,14 +360,77 @@ This does not go for bug reports however, as we want to unearth and solve as man
...
+
+You should now be ready to create a _good_ branch that follows our branch name policy. But if you are interested in why we have decided on this policy - read on. 
+
+#### Why this branch naming policy?
```
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:391]: https://github.com/kirbydesign/designsystem/pull/1490#discussion_r621084565

File: .github/CONTRIBUTING.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:374]: Aaand then they weren't... Seems like the [formatter](https://marketplace.visualstudio.com/items?itemName=mervin.markdown-formatter) inserts these for me 🧐

File: .github/CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkaltoft) [reply .github/CONTRIBUTING.md]: "Branch naming _manifest_" is my final suggestion... 😁

File: .github/CONTRIBUTING.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jkaltoft) [reply .github/CONTRIBUTING.md:366]: Hmmm, weird. Maybe we should see if there are other (better) formatters

**Review by jkaltoft — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md]: Manifest always sounds a bit _Karl Marxy_ to me 😅

File: .github/CONTRIBUTING.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (MadsBuchmann) [reply .github/CONTRIBUTING.md:366]: Yeah - think I managed to fix it for now tho 🤷‍♂️
Let's continue with the formatter we're using for now. In the end we should probably just run Prettier before merging `documentation/contribution-guidelines` , as prettier most likely will be the one future contributors be using, when editing this file 😄

**Pull request merged**